src/AppInstallerCLICore/ExecutionReporter.h

```diff
@@ -165,9 +165,30 @@ namespace AppInstaller::CLI::Execution
             m_progressSink = sink;
         }
 
+        void SetLevelEnabled(Level reporterLevel, bool setEnabled = true) {
+            const bool isEnabled = levelEnabled(reporterLevel);
+            // If the reporter is already in the state we want, do nothing
+            if (isEnabled == setEnabled)
+            {
+                return;
+            }
+            if (setEnabled) {
+                // Add the level to the list of enabled levels
+                m_enabledLevels.push_back(reporterLevel);
+            }
+            else {
+                // Remove the level from the list of enabled levels
+                m_enabledLevels.erase(std::remove(m_enabledLevels.begin(), m_enabledLevels.end(), reporterLevel), m_enabledLevels.end());
+            }
+        }
+
     private:
         Reporter(std::shared_ptr<BaseStream> outStream, std::istream& inStream);
 
+        bool levelEnabled(Level reporterLevel) {

```

Comment:
nit: code style: function name is PascalCase and our code block starts a new line with '{', applies to all new code added in this file
```suggestion
        bool LevelEnabled(Level reporterLevel)
        {

```